Framework/PythonInterface/mantid/plots/plotfunctions.py

```diff
@@ -263,12 +265,17 @@ def get_plot_fig(overplot=None, ax_properties=None, window_title=None, axes_num=
     :return: Matplotlib fig and axes objects
     """
     import matplotlib.pyplot as plt
+
     if fig and overplot:
         fig = fig
     elif fig:
         fig, _, _, _ = create_subplots(axes_num, fig)
     elif overplot:
+        assert not fig  # The create subplot below assumes no figure was passed in
```

Comment:
I don't think this `assert` needs to be here because `fig` is guaranteed to be false, since if it were true then the elif above would be executed instead.

Reply:
In this case the assert is to make sure the elif above doesn't get refactored in a way that breaks this assumption

Would a comment to that effect next to the assert help?